src/toolkit/gis.h

```diff
@@ -0,0 +1,158 @@
+#ifndef CYCLUS_SRC_GIS_H_
+#define CYCLUS_SRC_GIS_H_
+
+#include <math.h>
+#include <stdio.h>
+#include <sstream>
+#include <string>
+
+#include "cyclus.h"
+
+// forward declarations
+namespace cyclus {
+namespace toolkt {
+class GIS;
+}
+}
+
+namespace cyclus {
+namespace toolkit {
+/// @class GIS
+/// The GIS class is a basic class that stores the geographic location of each
+/// agent in latitude and longitude and follows the ISO
+/// 6709 standard. Allows string expression of GIS objects as specified in
+/// ANNEX H of ISO 6709 found here: 'https://www.iso.org/standard/39242.html'
+/// or a quick simplified version found here
+/// 'https://en.wikipedia.org/wiki/ISO_6709'
+///
+/// Brief Description of ISO 6709:
+/// 1. Latitude comes before Longitude
+/// 2. North latitude is positive
+/// 3. East longitude is positive
+/// 4. Fraction of degrees (decimal values) is prefered in digital data
+/// exchange.
+///
+/// The class is adapted from 'https://github.com/jaime-olivares/coordinate'
+/// under the MIT License.
+///
+/// Longitude and Latitude is stored as seconds of degrees. This was also
+/// adapted from 'https://github.com/jaime-olivares/coordinate'. This, according
+/// to the jaime-olivares, allows the coordinate elements such as degrees,
+/// minutes, and seconds to "remain on the integral portion of values, with the
+/// exception of decimal of seconds, avoiding loss of precision." This is
+/// calculated by multiplying decimal degrees by 3600.
+/// example: 05.2169 -> 18780.84
+
+class GIS {
+ public:
+  /// The default constructor for GIS. This will creat an object with
+  /// lat =0, and long=0.
+  GIS();
+
+  /// GIS constructor with latitude and longditude as degrees as a decimal
+  /// number.
+  /// @param deciaml_lat latitude expressed in degrees as a decimal number.
+  /// @param decimal_lon longditude expressed in degrees as a decimal number.
+  GIS(float decimal_lat, float decimal_lon);
```

Comment:
Having read through his justifications about using float over double, some of them seem off to me.

First, the point you reference about loss of precision is only relevant for floating point arithmetic. Floating points need to undergo an operation in order for them to lose precision due to underflow or overflow of values. However, this comment is references storage, where the precision is constant.  That is, if you are storing (in memory or on disk) a floating point value, the precision is fixed by the number of mantissa bits that you chose to store.  

On the website, he claims that

> [Floats have] half the space and half the time to load in memory [of doubles].

This is likely not true for the standard systems out there.  Load time is dependent on the default word size of the processor and the OS. [Disclaimer: you always need to profile, of course :)] For a 64 bit system, the word size is 64 bits.  So a float is going to load in the same time as a double. Depending on how memory is aligned, a float could take up 64 bits too, with the last 32 bits being unused.  Furthermore, for most 64 bit architectures, floating point arithmetic is faster for doubles than it is for floats.  This is because these systems have been optimized for doubles and `float` operations require a cast to double first!

The place where floats truly are half the size and half the time to load into memory of doubles is in large arrays numbers.  However, that is not the situation we have here, where we have a couple of data members attached to a class.  

The last point, 

> The precision is enough compared to GPS precision (5-10 meters). A longitude with a maximum value of 180 degrees at the Equator that has 60 nautical miles per minute. When stored in a float variable, will have a minimum precision of about 2.4 meters.

This is pretty valid.  I am not sure we are GPS focused, but the question becomes, "Is 2.4 meters enough?" I don't really know how to answer this because it seems very problem-dependent. However, it does seem that if we just went with doubles, we wouldn't have to worry about whether 2.4 meters is enough.  By analogy, the precision of doubles is 4.4 nm, much smaller than anything we can conceive of worrying about for the time being.